readme.md

```diff
@@ -160,6 +160,15 @@ following locations:
 - "Server Properties" setting for the Metals VS Code extension, which can be
   configured per-workspace or per-user.
 
+## Custom artifact repositories (Maven or Ivy resolvers)
```

Comment:
It's good to add documentation here, I will followup after this PR is merged to add the same section to the main Metals website. Before every new release of this extension, we copy-paste everything from the Metals website and replace this readme.